Main.cpp

```diff
@@ -133,27 +133,27 @@ ScenarioGUI *g_pScenarioGUI = 0;
 Controller *g_pMainMenuController = 0;
 
 enum StarSize {
-    StarSmall = 0,
-    StarLarge,
-    StarHuge,
+	StarSmall = 0,
+	StarLarge,
+	StarHuge,
 };
 
 struct Star {
-    // Bitmap representation
-    BITMAP *m_pBitmap;
-    // Center location on screen
-    Vector m_Pos;
-    // Bitmap offset
-//    int m_Offset;
-    // Scrolling ratio
-    float m_ScrollRatio;
-    // Normalized intensity 0-1.0
-    float m_Intensity;
-    // Type
-    StarSize m_Size;
-
-    Star() { m_pBitmap = 0; m_Pos.Reset(); m_ScrollRatio = 1.0; m_Intensity = 1.0; m_Size = StarSmall; }
-    Star(BITMAP *pBitmap, Vector &pos, float scrollRatio, float intensity) { m_pBitmap = pBitmap; m_Pos = pos; m_ScrollRatio = scrollRatio; m_Intensity = intensity; }
+	BITMAP *m_Bitmap = nullptr;
+	int m_PosX = 0;
+	int m_PosY = 0;
+	float m_ScrollRatio = 1.0F;
+	int m_Intensity = 0;
+	StarSize m_Size = StarSmall;
+
+	Star() {}
+	Star(BITMAP *pBitmap, int posX, int posY, float scrollRatio, int intensity)
```

Comment:
Why did you change this into an initialization list?

Reply:
That's just the modern way of doing it, why wouldn't I?

About the blinking star, good catch :) I accidentally made the intensity value overflow over 255, I'll get that fixed.

Reply:
C++98. *Modern*.
But ok I guess, there's some tiny performance benefit in doing an initialization list over regular assignment in the constructor body.
Just put the colon on the same line as the declaration
`Star(BITMAP *pBitmap, int posX, int posY, float scrollRatio, int intensity) :`